nengo_spinnaker/simulator.py

```diff
@@ -126,6 +126,36 @@ def __init__(self, network, dt=0.001, period=10.0, timescale=1.0):
         start = time.time()
         self.netlist = self.model.make_netlist(self.max_steps or 0)
 
+        # Create a controller for the machine and boot if necessary
+        self.job = None
+        if not use_spalloc:
+            # Use the specified machine rather than trying to get one
+            # allocated.
+            if hostname is None:
+                hostname = rc.get("spinnaker_machine", "hostname")
+        else:
+            # Attempt to get a machine allocated to us
+            from spalloc import Job
+
+            # Determine how many boards to ask for (assuming 16 usable cores
+            # per chip and 48 chips per board).
+            n_cores = sum(v.resources.get(Cores, 0) for v in
+                          self.netlist.vertices)
+            n_boards = int(np.ceil((n_cores / 16.) / 48.))
+
+            # Request the job
+            self.job = Job(n_boards)
+
+            # Wait until we're given the machine
+            logger.info("Waiting for machine allocation...")
+            self.job.wait_until_ready()
+
+            # Store the hostname
+            hostname = self.job.hostname
+
+        self.controller = MachineController(hostname)
+        self.controller.boot()
```

Comment:
Frustratingly, for whatever reason, in most Spalloc'd systems you need to wait a bit before actually attempting to boot an allocated machine. This appears to be an artefact of having more layers of network between you and the machine than usual causing ARP to fail or something... Not really got a good solution to this... In future Spalloc server versions a longer pause before reporting a board as up may be the solution...